src/result.ts

```diff
@@ -2,48 +2,61 @@ import { Maybe, nullable, Nothing } from './maybe'
 
 export type Result<O, E> = {
   ap: <A>(r: Result<A, E>) => Result<A, E>
-  map: <A>(cb: (arg?: O) => A) => Result<A, E>
-  mapErr: <A>(cb: (arg?: E) => A) => Result<O, A>
-  chain: <A>(cb: (arg?: O) => Result<A, E>) => Result<A, E>
-  chainErr: <A>(cb: (arg?: E) => Result<O, A>) => Result<O, A>
+  map: <A>(cb: (arg: O) => A) => Result<A, E>
+  mapErr: <A>(cb: (arg: E) => A) => Result<O, A>
+  chain: <A>(cb: (arg: O) => Result<A, E>) => Result<A, E>
+  chainErr: <A>(cb: (arg: E) => Result<O, A>) => Result<O, A>
   swap: () => Result<O, E>
-  bimap: <A, B>(ok: (arg?: O) => A, err: (arg?: E) => B) => Result<A, B>
+  bimap: <A, B>(ok: (arg: O) => A, err: (arg: E) => B) => Result<A, B>
   cata: <A, B>(obj: {
-    Ok: (arg?: O) => A
-    Err: (arg?: E) => B
-  }) => A|B
+    Ok: (arg: O) => A
+    Err: (arg: E) => B
+  }) => A | B
   toMaybe: () => Maybe<O>
   inspect: () => string
   isErr: () => boolean
   isOk: () => boolean
 }
 
-export const Ok = <O>(arg?: O): Result<O, any> => ({
+
+const innerOk = <O>(arg: O): Result<O, any> => ({
```

Comment:
Let's rename to `_Ok` to signify it's private
```suggestion
const _Ok = <O>(arg: O): Result<O, any> => ({
```

Reply:
Done.